### PR TITLE
Improve infinite scrolling and fix bugs

### PR DESCRIPTION
**Pull request by Reddiepoint**

### CODE REVIEW

**Comment by Reddiepoint** [comment]
Added function to partially resolve #28.

File: cs-rin-ru-enhanced-mod.user.js [hunk 1]
```diff
@@ -449,13 +462,26 @@ function hideScs() {
...
 function colorize(str) {
     let lstr = str.toLowerCase();
     let hash = 0;
     for (let i = 0; i < lstr.length; i++) {
         hash = lstr.charCodeAt(i) + ((hash << 5) - hash);
     }
-    const color = Math.floor(Math.abs((Math.sin(hash) * 10000) % 1 * 16777216)).toString(16);
+    let color = Math.floor(Math.abs((Math.sin(hash) * 10000) % 1 * 16777216)).toString(16);
+    let rgb = hexToRgb(color);
+    while (rgb[0] + rgb[1] + rgb[2] < 350) {
```
[the remark this answers is not in the document]
Comment (Altansar69) [reply cs-rin-ru-enhanced-mod.user.js:480]: Yes, you're right. You'd have to check the background color and make sure there's enough contrast.

[the remark this answers is not in the document]
Comment (Reddiepoint) [reply cs-rin-ru-enhanced-mod.user.js:480]: I'll have to figure out how to do that unless one of you guys helps me😅

File: cs-rin-ru-enhanced-mod.user.js [hunk 2]
```diff
@@ -449,13 +463,29 @@ function hideScs() {
...
     for (let i = 0; i < lstr.length; i++) {
         hash = lstr.charCodeAt(i) + ((hash << 5) - hash);
     }
-    const color = Math.floor(Math.abs((Math.sin(hash) * 10000) % 1 * 16777216)).toString(16);
+    let color = Math.floor(Math.abs((Math.sin(hash) * 10000) % 1 * 16777216)).toString(16);
+    let rgb = hexToRgb(color);
+    let matches = bgColour.match(/^rgb\((\d+),\s*(\d+),\s*(\d+)\)$/);
+    const bgRgb = [parseInt(matches[1]), parseInt(matches[2]), parseInt(matches[3])]
+    console.log(bgRgb)
+    while (Math.abs(rgb[0] + rgb[1] + rgb[2] - (bgRgb[0] + bgRgb[1] + bgRgb[2])) < 200) {
```
Comment (Reddiepoint) [inline cs-rin-ru-enhanced-mod.user.js:484]: I haven't decided on what to put for the threshold, right now I have it set to 200. Maybe something higher like 300 would make it look nicer on the light theme.

File: cs-rin-ru-enhanced-mod.user.js [hunk 3]
```diff
@@ -412,12 +425,13 @@ function tagify() {
     if (options.custom_tags) {
         $(".titles, .topictitle").each(function () {
             const titleElem = this;
+            const parentElem = titleElem.parentElement
```
Comment (SubZeroPL) [inline cs-rin-ru-enhanced-mod.user.js:428]: is it needed anywhere else? if not then maybe it would be better to just move it to colorize function

**Review by SubZeroPL — CHANGES_REQUESTED** [review]

**Review by SubZeroPL — DISMISSED** [review]

**Comment by Reddiepoint** [comment]
Uh oh, I think I did something wrong when I was trying to resolve the merging 😓

**Review by SubZeroPL — APPROVED** [review]

**Pull request merged**